File: client/src/services/progressService.js
import axiosClient from "../api/axiosClient";

export async function getProgress(sessionId) {

    const response = await axiosClient.get(
        `/progress/${sessionId}`
    );

    return response.data;
}

export async function updateProgress(data) {

    const response = await axiosClient.patch(
        "/progress",
        data
    );

    return response.data;
}

File: backend/src/main/java/com/careeros/service/ProgressService.java
package com.careeros.service;

import com.careeros.dto.request.UpdateProgressRequest;
import com.careeros.dto.response.ProgressResponse;

public interface ProgressService {

    ProgressResponse updateProgress(
            UpdateProgressRequest request
    );

    ProgressResponse getProgress(
            String sessionId
    );

}